docs/psoc6/quickref.rst

```diff
@@ -138,6 +138,66 @@ The following constants are used to configure the pin objects in addition to the
           
     Selects the pin value.
 
+
+Software I2C bus
+----------------
+Software I2C (using bit-banging) works on all output-capable pins, and is
+accessed via :ref:`machine.SoftI2C <machine.SoftI2C>` ::
+
+    from machine import Pin, SoftI2C
+
+    i2c = SoftI2C(scl='P5_0', sda='P5_1', freq=100000)
+
+    i2c.scan()              # scan for devices
+
+    i2c.readfrom(0x3a, 4)   # read 4 bytes from device with address 0x3a
+    i2c.writeto(0x3a, '12') # write '12' to device with address 0x3a
+
+    buf = bytearray(10)     # create a buffer with 10 bytes
+    i2c.writeto(0x3a, buf)  # write the given buffer to the peripheral
+
+
+Hardware I2C bus
+----------------
+Hardware I2C works on the following listed pair of I2C pins
+
+=====  ===========  ============
+\      Default  
+=====  ===========  ============
+scl    P6_0           P9_0
+sda    P6_1           P9_1
+=====  ===========  ============
+
+
+The driver is accessed via :ref:`machine.I2C <machine.I2C>`
+
+The constructor
+^^^^^^^^^^^^^^^
+An instance of the :mod:`machine.I2C` class can be created by invoking the constructor with all the 
+necessary parameters to fully configure the ``I2C``. By invoking the constructor I2C peripheral is 
+initialized and configured to work in master mode.
+
+i2c = I2C(0,scl='P6_0',sda='P6_1',freq=4000000)
+
+Here i2c_id=0 should be passed mandatorily which selects the master mode operation.
```

Comment:
As per the MPY docs, the parameter is called `id`. 

Given here: file://wsl.localhost/Ubuntu_old/home/boramonideep/work/micropython_fork_2/docs/build/html/library/machine.I2C.html#machine.I2C